File: 01-in-memory/cs/Controllers/InMemoryController.cs
using Microsoft.AspNetCore.Mvc;
using InMemory.Accumulators;

namespace InMemory.Controllers;

public class AccumulatorRequest
{
    public string action { get; set; }
    public int value { get; set; }
}

[ApiController]
public class InMemoryController : ControllerBase
{
    private Accumulator accumulator;

    public InMemoryController()
    {
        accumulator = Accumulator.GetInstance();
    }

    private int getValueQueryParam()
    {
        return Int32.Parse(HttpContext.Request.Query["value"]);
    }

    [HttpGet("/status")]
    public AccumulatorStatus GetAccumulatorStatus()
    {
        return accumulator.Get();
    }

    [HttpPost("/increment")]
    public void IncrementAccumulator()
    {
        accumulator.Inc(getValueQueryParam());
    }

    [HttpPost("/decrement")]
    public void DecrementAccumulator()
    {
        accumulator.Dec(getValueQueryParam());
    }

    [HttpPost("/accumulator")]
    public void AccumulateFromMessage([FromBody] AccumulatorRequest body)
    {
        if (body.action == "increment")
            accumulator.Inc(body.value);

        if (body.action == "decrement")
            accumulator.Dec(body.value);
    }
}

File: 01-in-memory/kotlin/src/main/kotlin/com/upsetbit/kotapi/inmemory/InMemoryController.kt
package com.upsetbit.kotapi.inmemory

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

import com.upsetbit.kotapi.inmemory.accumulator.accumulator
import com.upsetbit.kotapi.inmemory.accumulator.AccumulatorStatus

@RestController
class InMemoryController
{
    @GetMapping("/status")
    fun getAccumulatorStatus() = accumulator.get()

    @PostMapping("/increment")
    fun incrementAccumulator(@RequestParam value: Int) = accumulator.inc(value)

    @PostMapping("/decrement")
    fun decrementAccumulator(@RequestParam value: Int) = accumulator.dec(value)

    @PostMapping("/accumulator")
    fun accumulateFromMessage(@RequestBody body: AccumulatorRequest)
    {
        if (body.action == "increment")
            accumulator.inc(body.value)

        if (body.action == "decrement")
            accumulator.dec(body.value)
    }
}
